.ci/ci-fast-return.sh

```diff
@@ -0,0 +1,183 @@
+
+	# do we have any patterns to check?
+	if [ -n "$mustcheck" ]; then
+		local need_checking=$(egrep ${mustcheck[@]} <<< "$files" || true)
+	else
+		info "No force CI check patterns" >&2
+		local need_checking=""
+	fi
+
+	# If we have any files that *must* be checked, then immediately return
+	if [ -n "$need_checking" ]; then
+		# stderr, so it does not get in our return value...
+		cat >&2 <<-EOT
+		INFO: Cannot fastpath skip CI.
+		INFO: Some files present must be CI checked.
+		INFO: Files to check are:
+
+		$need_checking
+
+EOT
+		echo "1"
+		return
+	else
+		info "No force check files found" >&2
+	fi
+
+	# Now we have checked there are no 'must check' files, we can check to see
+	# if all files fall into the 'can skip' patterns.
+	# first, do we have any skip patterns to search for?
+	if [ -n "$canskip" ]; then
+		local non_skippable=$(egrep -v ${canskip[@]} <<< "$files" || true)
+	else
+		info "No skip CI check patterns" >&2
+		# No patterns, set non-skippable list to all files then...
+		local non_skippable="$files"
+	fi
+
+	if [ -z "$non_skippable" ]; then
+		# stderr, so it does not get in our return value...
+		cat >&2 <<-EOT
+		INFO: No files to check in CI.
+		INFO: Fastpath short circuit returning from CI.
+		INFO: Files skipped are:
+
+		$files
+
+EOT
+		echo "0"
+		return
+	else
+		cat >&2 <<-EOT
+		INFO: Not all files skippable
+
+		$non_skippable
+
+EOT
+		echo "1"
+		return
+	fi
+}
+
+# Check if we have the 'magic label' that forces a CI run set on the PR
+# Return:
+#  0 - No label found, could skip the CI
+#  1 - Label found - should run the CI
+check_force_label() {
+
+	[ -z "$ghprbGhRepository" ] && info "No ghprbGhRepository set, skip label check" >&2 && echo "0" && return
+	[ -z "$ghprbPullId" ] && info "No ghprbPullId set, skip label check" >&2 && echo "0" && return
+
+	info "Checking labels for PR ${ghprbGhRepository}/${ghprbPullId}" >&2
+
+	# Pull the label list for the PR
+	# Ideally we'd use a github auth token here so we don't get rate limited, but to do that we would
+	# have to expose the token into the CI scripts, which is then potentially a security hole.
+	# json=$(curl -H "Authorization: Basic ${GITHUB_API_TOKEN}" https://api.github.com/repos/${ghprbGhRepository}/issues/${ghprbPullId}/labels)
+	local json=$(curl -s https://api.github.com/repos/${ghprbGhRepository}/issues/${ghprbPullId}/labels)
+
+	# Pull the label list out
+	local labels=$(jq .[].name <<< $json)
+
+	# Check if we have the forcing label set
+	for x in $labels; do
+		# Strip off any surrounding '"'s
+		y=$(sed 's/"//g' <<< $x)
+		[ "$y" == "$force_label" ] && info "label found, forcing CI" >&2 && echo "1" && return
+	done
+
+	info "No forcing label found" >&2
+	echo "0"
+}
+
+main()
+{
+	info "Checking for any changed files that will prevent CI fastpath return"
+	res=$(can_we_skip)
+
+	# If the file check says we can skip, check the labels to see if we are forcing the
+	# CI run anyway.
+	[ $res == 0 ] && res=$(check_force_label)
```

Comment:
Nit: I know this works, but I think this should be using `-eq` for comparing numerics?